storage/testonly/fake_node_reader.go

```diff
@@ -46,21 +39,19 @@ type FakeNodeReader struct {
 	nodeMap      map[string]storage.Node
 }
 
-// NewFakeNodeReader creates and returns a FakeNodeReader with the supplied nodeID -> Node
-// mappings assuming that all the nodes are at a specified tree revision. All the nodeIDs
+// NewFakeNodeReader creates and returns a FakeNodeReader with the supplied nodes
+// assuming that all the nodes are at a specified tree revision. All the node IDs
 // must be distinct.
-func NewFakeNodeReader(mappings []NodeMapping, treeSize, treeRevision int64) *FakeNodeReader {
+func NewFakeNodeReader(nodes []storage.Node, treeSize, treeRevision int64) *FakeNodeReader {
 	nodeMap := make(map[string]storage.Node)
 
-	for _, mapping := range mappings {
-		_, ok := nodeMap[mapping.NodeID.String()]
-
-		if ok {
-			// Duplicate mapping - the test data is invalid so don't continue
-			glog.Fatalf("NewFakeNodeReader duplicate mapping for: %s in:\n%v", mapping.NodeID.String(), mappings)
+	for _, node := range nodes {
+		id := node.NodeID.String()
```

Comment:
It's generally not safe to rely on string representations as IDs, which is why that mapping exists. As this is test code it might be acceptable.

Reply:
The previous code also used the `String()` representation as a hashmap key, this PR doesn't change that.

Reply:
Yeah. It wasn't a great idea then either. I suppose we do at least define String() on this to return the binary value.